<robot name="vc_robot_v1"
       xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:include filename="$(find beacon_gazebo_sim)/urdf/vc_robot_v1/vc_robot_v1_gazebo.xacro" />
    <xacro:include filename="$(find beacon_gazebo_sim)/urdf/vc_robot_v1/macros.xacro" />



    <xacro:property name="pi" value="3.1415926535897931"/>

    <link name="base_link">
        <!-- pose and inertial -->
        <pose>0 0 0.05 0 0 0</pose>
        <inertial>
            <mass value="0.5"/>
            <origin rpy="0 0 0" xyz="0 0 0.05"/>
            <inertia ixx="0.206208333333" ixy="0" ixz="0" iyy="0.206208333333" iyz="0" izz="0.206208333333"/>
        </inertial>
        <!-- body -->
        <collision name="collision_chassis">
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
<!--                <box size="0.295 0.295 0.05"/>-->
                <cylinder length="0.07" radius="${0.345/2}"/>
            </geometry>
        </collision>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
<!--                <box size="0.295 0.295 0.05"/>-->
                <cylinder length="0.07" radius="${0.345/2}"/>
            </geometry>
            <!--      <material name="blue"/>-->
        </visual>
        <!-- caster front -->
        <collision name="caster_front_collision">]
            <origin rpy="0 1.5707 0" xyz="0.05 0 -0.012"/>
            <geometry>
                <cylinder length="0.025" radius="0.038"/>
            </geometry>
            <surface>
                <friction>
                    <ode>
                        <mu>0</mu>
                        <mu2>0</mu2>
                        <slip1>1.0</slip1>
                        <slip2>1.0</slip2>
                    </ode>
                </friction>
            </surface>
        </collision>
        <!-- <visual name="caster_front_visual">
          <origin rpy="0 1.5707 0" xyz="0.05 0 -0.012"/>
          <geometry>
            <cylinder length="0.025" radius="0.038"/>
          </geometry>
        </visual> -->
        <collision name="caster_back_collision">]
            <origin rpy="0 1.5707 0" xyz="-0.05 0 -0.012"/>
            <geometry>
                <cylinder length="0.025" radius="0.038"/>
            </geometry>
            <surface>
                <friction>
                    <ode>
                        <mu>0</mu>
                        <mu2>0</mu2>
                        <slip1>1.0</slip1>
                        <slip2>1.0</slip2>
                    </ode>
                </friction>
            </surface>
        </collision>
        <visual name="caster_back_visual">
            <origin rpy="0 1.5707 0" xyz="-0.05 0 -0.012"/>
            <geometry>
                <cylinder length="0.025" radius="0.038"/>
            </geometry>
        </visual>
    </link>

    <!-- <link name="sensor_laser">
      <inertial>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <mass value="1" />
        <xacro:cylinder_inertia mass="1" r="0.05" l="0.1" />
      </inertial>

      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <cylinder radius="0.05" length="0.1"/>
        </geometry>
        <material name="white" />
      </visual>

      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <cylinder radius="0.05" length="0.1"/>
        </geometry>
      </collision>
    </link> -->

    <!-- <joint name="joint_sensor_laser" type="fixed">
      <origin xyz="0.15 0 0.05" rpy="0 0 0"/>
      <parent link="link_chassis"/>
      <child link="sensor_laser"/>
    </joint>
     -->
    <xacro:link_wheel name="link_right_wheel" />
    <xacro:joint_wheel name="joint_right_wheel" child="link_right_wheel" origin_xyz="0 0.16 -0.012" parent="base_link" />

    <xacro:link_wheel name="link_left_wheel" />
    <xacro:joint_wheel name="joint_left_wheel" child="link_left_wheel" origin_xyz="0 -0.16 -0.012" parent="base_link" />

    <joint name="front_bumper_joint" type="fixed">
        <axis xyz="0 1 0" />
        <origin xyz="0.2 0 0" rpy="0 0 0"/>
        <parent link="base_link"/>
        <child link="link_front_bumper"/>
    </joint>
    <!--  <gazebo reference='front_bumper_joint'>-->
    <!--    <preserveFixedJoint>true</preserveFixedJoint>-->
    <!--  </gazebo>-->
    <joint name="imu_joint" type="fixed">
        <axis xyz="0 1 0" />
        <origin xyz="0 0 0.01" rpy="0 0 0"/>
        <parent link="base_link"/>
        <child link="imu_link"/>
    </joint>
    <gazebo reference='imu_joint'>
        <preserveFixedJoint>true</preserveFixedJoint>
    </gazebo>
    <joint name="receiver_joint" type="fixed">
        <axis xyz="0 1 0" />
        <origin xyz="0 0 0.05" rpy="0 0 0"/>
        <parent link="base_link"/>
        <child link="receiver_link"/>
    </joint>
    <gazebo reference='receiver_joint'>
        <preserveFixedJoint>true</preserveFixedJoint>
    </gazebo>
    <joint name="beacon_joint" type="fixed">
        <axis xyz="0 1 0" />
        <origin xyz="0 0 0.1" rpy="0 0 0"/>
        <parent link="base_link"/>
        <child link="beacon_link"/>
    </joint>
    <gazebo reference='beacon_joint'>
        <preserveFixedJoint>true</preserveFixedJoint>
    </gazebo>
    <!--  &lt;!&ndash; Camera &ndash;&gt;-->
    <link name="link_front_bumper">
        <pose>0 0 0 0 0 0</pose>
        <collision name="link_front_bumper_collision">
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <box size="0.04 0.20 0.04"/>
            </geometry>
        </collision>

        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <box size="0.04 0.20 0.04"/>
            </geometry>
            <!--      <material name="red"/>-->
        </visual>

        <inertial>
            <mass value="1e-5" />
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
        </inertial>
    </link>

    <link name="imu_link">
        <pose>0 0 0 0 0 0</pose>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <box size="0.04 0.04 0.02"/>
            </geometry>
        </collision>

        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <box size="0.04 0.04 0.04"/>
            </geometry>
            <!--      <material name="green"/>-->
        </visual>

        <inertial>
            <mass value="1e-5" />
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
        </inertial>
    </link>

    <link name="receiver_link">
        <pose>0 0 0 0 0 0</pose>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <box size="0.02 0.02 0.04"/>
            </geometry>
        </collision>

        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <box size="0.02 0.02 0.04"/>
            </geometry>
            <!--      <material name="green"/>-->
        </visual>
        <inertial>
            <mass value="1e-5" />
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
        </inertial>
    </link>

    <link name="beacon_link">
        <pose>0 0 0 0 0 0</pose>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <box size="0.02 0.02 0.04"/>
            </geometry>
        </collision>

        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <box size="0.02 0.02 0.04"/>
            </geometry>
            <!--      <material name="green"/>-->
        </visual>
        <inertial>
            <mass value="1e-5" />
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
        </inertial>
    </link>
</robot>